applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py

```diff
@@ -329,6 +329,13 @@ def AddVariables(self):
 
         KratosMultiphysics.Logger.PrintInfo("NavierStokesEmbeddedMonolithicSolver", "Fluid solver variables added correctly.")
 
+    def AddDofs(self):
+        super(NavierStokesEmbeddedMonolithicSolver, self).AddDofs()
+        if self.__fm_ale_is_active:
+            KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.MESH_DISPLACEMENT_X, KratosMultiphysics.MESH_REACTION_X, self.main_model_part)
+            KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.MESH_DISPLACEMENT_Y, KratosMultiphysics.MESH_REACTION_Y, self.main_model_part)
+            KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.MESH_DISPLACEMENT_Z, KratosMultiphysics.MESH_REACTION_Z, self.main_model_part)
```

Comment:
@rubenzorrilla I think you need to add the dofs for `MESH_DISPLACEMENT` if you use the fm_ale
In the embedded piston test you are explicitly fixing those dofs

Let me know what you think

Reply:
The `MESH_DISPLACEMENT` DOFs are added in the virtual model part, not in the fluid one. They are added in lines 117-119 of fixed_mesh_ale_utilities.cpp.

This should not be changed.

Reply:
I mean, we can add them in here too, but I prefer to keep all the FM-ALE stuff in a unique place.

Reply:
Makes sense to me
But then can you please change the test to fix those dofs in the virtual and not in the fluid ModelPart?

Reply:
I checked your implementation, but I don't understand what you are doing.
The virtual and the fluid -MP have DIFFERENT nodes, i.e. fixing the `MESH_DISPLACEMENT` dofs in the virtual ModelPart won't work I think :thinking: 

Can you please check it yourself?

Reply:
The virtual model part nodes are created as a copy of the fluid model part one (same id and coordinates). Indeed the idea is to fix the ALE DOFs only in the virtual model part.

Reply:
Now I got your point... I thought you were adding the nodes in the fluid model part to do the mesh problem. The point is that we set the fixity in the virtual model part by copying the one of the fluid model part... This was somehow hidden until this issue was raised...

At this point, I see two clean solutions:
1. The virtual model part nodes are not a copy but shared with the fluid model part. I think this can't be achieved as we need to do some operations with the database when doing the FM-ALE projections.
2. When filling the virtual model part, we also copy the submodelparts from the fluid origin model part. Then in the json it would be possible to apply the fixity as `VirtualModelPart.fixed_mesh_displacement_sub_model_part` instead of `FluidModelPart.fixed_mesh_displacement_sub_model_part`.

Reply:
my main confusion is that you create the VirtualMP with the mainFluid-MP
However then you fix nodes in SUBModelParts of the FluidMP, that do not exist in the VirtualMP
I.e. I thought the mesh-movement is done in the VirtualMP right? But how does it work then, since you don't fix the dofs in the VitrualMP
is this a bug?

As I said, I don't get it and hence would appreciate if you could take a look yourself

Reply:
More than a bug, I'd say it is a misuse of what @emiroglu came across with. I was creating useless `MESH_DISPLACEMENT` DOFs in the fluid model part just to set the fixity. Then I was creating the `MESH_DISPLACEMENT` DOFs we use to solve the mesh problem and copying the fixity from the fluid virtual model part.

Reply:
> 
> As I said, I don't get it and hence would appreciate if you could take a look yourself

Sure! Thanks for spotting it.

The point is that I don't know how to proceed... I think we should firstly fix this before merging this PR. The point is that it wouldn't be a FastPR anymore.

Reply:
> I think that we should open a new branch, fix the FM-ALE stuff and then merge it into this one.

I agree!

> When filling the virtual model part, we also copy the submodelparts from the fluid origin model part. Then in the json it would be possible to apply the fixity as VirtualModelPart.fixed_mesh_displacement_sub_model_part instead of FluidModelPart.fixed_mesh_displacement_sub_model_part.

I gess thats what you have to do if you need a completely separate MP :/
Maybe you can use the ConnectivityPreserveModeler?

Reply:
Would be nice but I cant, as I need to have two independent nodal databases...

Reply:
@philbucher I was thinking in a more efficient way to do this but it would be a complete mess from the GUI point of view.

Having said this, the fix you did I think is the best we can do.

Please go ahead and many thanks!